### PR TITLE
test(bench): cover the harness's process half with a stand-in server

### PR DESCRIPTION
**Pull request by jmrplens**
`cmd/bench_resources` sat at 68% coverage, and Sonar put the new code of the benchmark PR at 72.5%, under the 80% gate. Everything below that line is the half of the harness that starts a process: the readiness poll, the pipe wiring, the resident-set sampler, the ramp, the load rounds and the traceback signal. None of it can be exercised against anything in-process, and linking the real server for it would cost a minute per test run to learn nothing about the harness. I preferred covering it to excluding it.

## The stand-in

`cmd/bench_resources/testdata/standin` is a Go program, standard library only, that answers the harness's contract in two hundred lines: the flags the HTTP target passes, the environment the stdio target sets (it refuses to start without `GITLAB_URL` and `GITLAB_TOKEN`, so a harness that stopped setting them is caught), a `/health` document, JSON-RPC over SSE on `/mcp` with the `Mcp-Method` and `PRIVATE-TOKEN` headers checked, newline-delimited JSON-RPC on stdio, and a runtime that dumps its goroutines on SIGQUIT because nothing in it catches the signal. `STANDIN_FAIL` names one method it refuses with an error result, which is how the tests reach the ramp's fatal path and the load phase's failure notes without a server that is actually broken. It follows `cmd/server/testdata/peer`: built by `TestMain` once per test binary, and outside Sonar's coverage like every `testdata` program.

## What is now tested

- Both targets end to end: readiness through `/health`, one process for every credential on HTTP and one process per client on stdio, the traceback that ends a target and a `close` that finds nothing left to stop, and the exec failure each reports.
- `runScenario` on both transports, including telemetry on: every startup milestone, the idle and per-client resident sets read from the kernel, a ramp point per credential, one distribution per method with every call counted, the goroutine count, and stdio's honest zero for readiness and idle memory.
- The failure branches: a refused cold `tools/list` fails the scenario; refused `tools/call`s are noted and the other distributions survive.
- `measure` over the quick matrix and `execute` measure → render → check over a temporary tree, which is the round trip `make` runs.
- `buildServer` against a throwaway module with a trivial `cmd/server`, so the real build path runs in a second, plus its failure.
- Small leftovers: `commandProcess` refusing pre-wired pipes, `firstLine`, `shortCommit`, the `n/a` table cells, `writeRun`'s two filesystem refusals, `rel` on a path that cannot be made relative.

Coverage goes from 68.5% to 93.1% (Go statements). Sonar's check on this PR reads "0.0% Coverage on New Code" because the PR adds no coverable line: it is tests plus a `testdata` program, which `sonar.exclusions` keeps out as a fixture, like `cmd/server/testdata/peer`. The figure that moved is the package's line coverage in the same analysis, 72.5% on main to 91.1% here, and the project's 97.5%. What remains is the darwin branches of the host probes, the non-Linux branch of the process reader, and the sixty-second health deadline. The process-level tests skip on Windows, where the harness has neither `/proc`, `ps` nor SIGQUIT to work with; the package's whole suite runs in about two seconds, three shuffled runs in five.

## Summary by Sourcery

Raise benchmark harness coverage by testing its process-based execution paths with a lightweight HTTP and stdio stand-in server.

Bug Fixes:
- Ensure command process setup cleans up an already-created stdin pipe when stdout pipe creation fails.

Enhancements:
- Exercise the benchmark harness end to end for HTTP and stdio process targets, including readiness, RPC handling, resource sampling, telemetry, ramping, tracebacks, shutdown, and failure reporting.
- Add a lightweight stand-in server fixture that supports the harness's HTTP and stdio contracts without requiring the real server.
- Cover benchmark measurement, rendering, validation, server building, result writing, formatting, path handling, and command-process error paths.
- Raise cmd/bench_resources coverage above 90% and remove it from the below-target package list.

Build:
- Build the stand-in server once per test binary and validate the real server build path with a temporary Go module.

Documentation:
- Update repository and testing metrics to reflect the added fixture, tests, package count, and improved coverage.

Tests:
- Add process-level integration tests for both benchmark transports and their success and failure branches.
- Add coverage for benchmark command round trips and remaining filesystem, formatting, path, and pipe-handling edge cases.

## Summary by CodeRabbit

* **Documentation**
  * Updated README statistics to reflect the current codebase.
  * Refreshed testing metrics, including test counts and coverage figures.

* **Tests**
  * Expanded benchmark resource testing across HTTP and stdio workflows.
  * Added coverage for process lifecycles, error handling, rendering, measurements, and failure scenarios.
  * Added a standalone test server to support end-to-end benchmark validation.

### CODE REVIEW

File: cmd/bench_resources/main_test.go [hunk 1]
```diff
@@ -248,3 +253,212 @@ func TestOptionsValidate_RejectsValuesThatWouldMeasureNothing(t *testing.T) {
...
+func runWithStandin(m *testing.M) int {
+	if runtime.GOOS != "windows" {
+		dir, err := os.MkdirTemp("", "bench-standin")
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "create a build directory for the stand-in: %v\n", err)
+			return 1
+		}
+		defer func() { _ = os.RemoveAll(dir) }()
+		path := filepath.Join(dir, "standin")
+		build := exec.CommandContext(context.Background(), "go", "build", "-o", path, "./testdata/standin")
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply cmd/bench_resources/main_test.go]: Applied. The stand-in build now runs under a ten-minute context, the bound buildServer already uses, cancelled when runWithStandin returns.

File: cmd/bench_resources/rpc_test.go [hunk 2]
```diff
@@ -322,3 +323,44 @@ func TestStdioRPC_ContextCancelled_ReturnsPromptly(t *testing.T) {
...
+// TestCommandProcess_RefusesPipesAlreadyWired covers the two ways exec.Cmd
+// declines to hand out a pipe: a stream the caller already assigned.
+func TestCommandProcess_RefusesPipesAlreadyWired(t *testing.T) {
+	cases := []struct {
+		name    string
+		prepare func(*exec.Cmd)
+		wantErr string
+	}{
+		{name: "stdin already set", prepare: func(c *exec.Cmd) { c.Stdin = strings.NewReader("") }, wantErr: "stdin pipe"},
+		{name: "stdout already set", prepare: func(c *exec.Cmd) { c.Stdout = io.Discard }, wantErr: "stdout pipe"},
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply cmd/bench_resources/rpc_test.go:336]: Applied. commandProcess closes the stdin pipe when StdoutPipe fails: the command is never started on that path, so nothing else would have closed it.

File: cmd/bench_resources/run_test.go [hunk 3]
```diff
@@ -365,3 +365,174 @@ func TestDecodeHealth_ReadsTheBuildTheServerReports(t *testing.T) {
...
+	}
+	wantNote := strconv.Itoa(plan.Clients*plan.Parallel) + " tools/call calls failed"
+	if len(scenario.Notes) != 1 || !strings.Contains(scenario.Notes[0], wantNote) {
+		t.Errorf("notes %v, want one saying %q", scenario.Notes, wantNote)
+	}
+	if calls, ok := scenario.latency(methodToolsCall); !ok || calls.Count != 0 {
+		t.Errorf("tools/call distribution %+v, want an empty one kept in place", calls)
+	}
+	if list, ok := scenario.latency(methodToolsList); !ok || list.Count != plan.Clients*plan.Parallel {
+		t.Errorf("tools/list distribution %+v, want unaffected", list)
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply cmd/bench_resources/run_test.go:542]: Applied. The failed-calls test now asserts the resources/list distribution kept every call as well, beside tools/list.

File: cmd/bench_resources/testdata/standin/main.go [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+	Message string `json:"message"`
+}
+
+func main() {
+	httpMode := flag.Bool("http", false, "serve HTTP instead of stdio")
+	addr := flag.String("http-addr", "", "listen address in HTTP mode")
+	flag.String("gitlab-url", "", "accepted and ignored")
+	flag.String("tool-surface", "", "accepted and ignored")
+	flag.Int("rate-limit-rps", 0, "accepted and ignored")
+	flag.Bool("telemetry", false, "accepted and ignored")
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply cmd/bench_resources/testdata/standin/main.go]: Applied. With --telemetry the stand-in posts one export to OTEL_EXPORTER_OTLP_ENDPOINT/v1/metrics at startup, best effort the way an exporter is, and TestRunScenario_HTTP_FillsEveryPublishedFigure asserts the sink received it, so a harness that stopped wiring the endpoint is caught.
